src/player.rs

```diff
@@ -234,13 +230,13 @@ impl GameInfo {
 
         self.last_seen += 1;
         if self.last_seen > DISCONNECTED_THRESHOLD {
-            self.disconnected = true;
+            self.state = PlayerState::Disconnected;
         }
     }
 
     pub(crate) fn acknowledge(&mut self) {
         self.last_seen = 0;
-        self.disconnected = false;
+        self.state = PlayerState::Active;
```

Comment:
Considering this could be set to either `Status::Spawning` or `Status::Active` instead, it should probably only be set if it was previously `Status::Disconnected`. I also think I would prefer it being set to `Spawning` instead of `Active` as a "default" state, although perhaps it doesn't really matter.

Reply:
Solved this in the latest commit